presto-hive/src/main/java/com/facebook/presto/hive/HivePageSource.java

```diff
@@ -230,22 +242,45 @@ public boolean isFinished()
     @Override
     public Page getNextPage()
     {
+        if (filterOnPrefilledValuesFailed) {
+            return null;
+        }
+
         try {
             Page dataPage = delegate.getNextPage();
             if (dataPage == null) {
                 return null;
             }
+
+            if (dataPage.getPositionCount() > 0 && filterOnPrefilledValuesException.isPresent()) {
```

Comment:
Hmm so I wonder why we would have a prefilled value exception if we didn't have any page positions?

Reply:
@tdcmeehan Tim, consider the following query 

`select * from t where a > 0 and cast(ds as integer) > 0`

Table t has 2 columns: a is a regular column, ds is a partition column. `cast(ds as integer)` expression throws an exception because ds is a '2019-01-02' string. If table t has rows with a > 0, then the query should fail. Otherwise, it should succeed.

In the latter case, early evaluation of `cast(ds as integer)` produces an error, however, that error needs to be surfaces only if `a > 0` returns some rows. The error should be suppressed if `a > 0` returns nothing.

Reply:
Makes sense, thanks for the explanation.